authoring-service/src/main/kotlin/com/devotion/authoring/dto/events.kt

```diff
@@ -1,13 +1,33 @@
 package com.devotion.authoring.dto
 
 import com.devotion.authoring.NoArgConstructor
+import com.fasterxml.jackson.annotation.JsonTypeInfo
 
-enum class Action {
-    CREATE, UPDATE, DELETE
-}
+@NoArgConstructor
+@JsonTypeInfo(use= JsonTypeInfo.Id.CLASS, include= JsonTypeInfo.As.PROPERTY, property="class")
+abstract class QuestionEvent( val surveyId: String, open val question: QuestionText)
```

Comment:
I hate inheritance. I'd rather use interface (`Kotlin` interfaces can hold data). Maybe you could try making this an `Interface`?